### ISSUE TITLE
stackoverflow toxicity: reacting to "Is Stack OverFlow Evil? | Prime React" reacting to "Uncovering Stack Overflow's TOXICITY"

### ISSUE THREAD

**Issue by deric4** [deric4/deric4.github.io#3]
### TL;DR

StackOverflow is just as toxic as solo queuing a hot drop with a stack of Apex Legends sweats... but I stil play

---

Saw a new video from the @ThePrimeagen and felt attacked because I've recently started answering questions on the toxic site.

In his video (linked below) he asks 

> Why do you answer questions and participate in stackoverflow

and

> How do you feel about gamification

after talking about the type of people who like to answer questions 😂 (which I pretty much agree with) but got triggered enough to write this... 

### some context:

* Quit my job as a Sr. Platform Engineer on my birthday Jan 2022 to go fuck around and play Apex Legends all day.

* I've been a SO user for over 8 years at this point, but aside from a few cringey questions early on in my career, almost all of my interaction has been in the last year, with most of it in Q2 2022.

  <p align="center">
  <img width="500" alt="summary" src="https://user-images.githubusercontent.com/5762138/233123429-9b42c6ea-dbeb-4b5a-b5e8-97b956a8a0dc.png">
  <p>

* I haven't really written any code on a computer since quitting, >90% has been on my  iPad/iPhone using  the browser editor/codespaces/AWS Lambda Editor/AWS cloudshell. i.e. [deric4/github-action-runner-reference](https://github.com/deric4/github-action-runner-reference). It sucks but not as bad as setting up a dev environment for arm64 that I didn't hate:
     
    ```
    # see
    # https://github.com/deric4/github-action-runner-reference/blob/main/.github/workflows/runner-reference-python.yml#L209-L214
         #   See Also: see https://gist.github.com/deric4/af90cc848c4915df803ce33ab1e5356e
         #   (some notes reminding me I was too stupid to figure out how/what needed to change for minor python version
         #   to be scoped in pre-commit virtualenv for python hooks (was having probs with black and developing a custom hook tolint changes
         #   to a geojson file to enforce right-hand-rule https://datatracker.ietf.org/doc/html/rfc7946#section-3.1.6 
         #   and validating projection conversions. Most of the modules I tested required running with `python3-intel64` on my local M1. Ended up implementing a pure python implementation
         #   but... I don't really trust it.
    ```

## as to why

Even though I haven't been working, i've been curious™️ about languages/frameworks/tools that I've never used, which has inevitably lead to a higher proportion of google search results with links to SO. My questions have been _so basic_ at times, search results from GitHub issues (where everyone should start looking) aren't helpful in a lot cases.

It got to the point where I was getting redirected to SO often enough, the stupid fucking cookie permissions banner made it worth actually logging in so it would remember my response. Another plus was that it remembered my dark mode preferences no longer melting my face every time I got redirected.

I got interested into _actively_ trying to answer questions around the time Advent of Code 2022 was about to start and started thinking about how much harder it would be doing it only from my iPad. Went down a YouTube rabbit hole of people solving past solutions and discovered one from AoC 2021 by Russ Cox (of Go)

<a href="https://www.youtube.com/watch?v=o6GoqXcXsp0" target="_blank"> 
 <h3>AoC 2021 Day 19 using Ivy [Aligning Scanners]</h3>
 <img width=200 src="https://i.ytimg.com/vi/o6GoqXcXsp0/hqdefault.jpg" alt="Watch the video" />
<a/>

I'm not sure that I had ever watched someone live code a solution like that and break it down in plain english, all the while just throwing straight sorcery into the terminal/REPL 🤯 . Its long (25:28) but I still don't understand how this only has 400 views.

It truly made think to myself, "Is there _anything_ that I'm as proficient in as that dude", obviously not, but I decided to experiment by "gamifying" answering stack overflow answers.

I put an artificial constraint on myself that the only time I could look at documentation (if needed) was at the end of my answer before submitting to check and provide useful info to the person asking. The workflow went pretty much like this:

1. Create a custom filter of tags for topics I wanted to focus in order 
2. If I found my self scrolling reddit, hop over to SO when I was done w/ reddit and scroll through the first 3-5 pages of most recent questions that I liked
3. Book mark any that I would have chosen if it already had an answer
4. If I found one i liked, open up my Notes app and raw dog a solution in plain text
5. If I was right, check answer against docs, post if correct, keep just for me if i was wrong to dive into later

And this was fun for me and I think at least some of my answers were truly helpful to someone else.

Maybe this isn't the same type of "gamification" that Prime is referring to when saying (paraphrasing here, not exact quote):

>  I dont care about gamification, there's nothing to it, pointless

which I 100% agree with, I could give a shit about the official point system from SO. 

I quit a 6 figure job because I'd rather go ski/snowboard for couple weeks at time, or chase interesting rabbit holes on weird error my code was hitting, or just sit on my ass playing Apex Legend... because the money wasn't why I was good at what I do. 

I _DO_ think the underlying upvote/downvote system is _useful enough_ to increase searchabilty/quality though. And its not like GitHub issues/discussions don't suffer from the same type of thing, try asking a question or reporting a problem in homebrew/homebrew-core after an update breaks your shit lol. 

Ultimately, I'm of the opinion SO is a toxic place for people that are "newbs", and especially so ones that may be there first time participating in an online forum and haven't been exposed to "the culture", but they ended up there because all their other google search results were spammy tutorials filled with ads. Whats the alternative? 

Where else can these beginner type questions be asked? It's not google, because search results are increasingly getting worse!

### Just google `python string length`!!!

The official python docs aren't even on the first fucking page of my results, and its been this way for years!

---

I'm not sure whats proper etiquette for attribution on social media platforms, but here's the links to the YouTube videos. I encourage yall to watch if you've read up to this point, both creators make entertaining content.

🙄 looks like GitHub removed the `target="_blank"` functionality when rendering forcing you to right click open new tab. 

<a href="https://www.youtube.com/watch?v=MCQWfMg0Z9w" target="_blank">
 <h3>Is Stack OverFlow Evil? | Prime Reacts</h3>
 
 <img width=300 src="https://i.ytimg.com/vi/MCQWfMg0Z9w/mqdefault.jpg" alt="Watch the video" />

<a href="https://www.youtube.com/watch?v=N7v0yvdkIHg" target="_blank">
 <h3>Uncovering Stack Overflow's TOXICITY</h3>
 
 <img width=300 src="https://i.ytimg.com/vi/N7v0yvdkIHg/mqdefault.jpg" alt="Watch the video" />

**Comment by deric4** [comment]
lol

> ## Just google python string length!!!
>
>The official python docs aren't even on the first fucking page of my results, and its been this way for years!

preview on the new OverflowAI splash page

https://stackoverflow.co/labs/

![15E592D9-9A40-4C1F-A739-BB402E266A7F](https://github.com/deric4/deric4.github.io/assets/5762138/8588a755-746a-424e-a8d2-487ac628459a)
